modules/internal/ChapelRange.chpl

```diff
@@ -2601,10 +2684,12 @@ operator :(r: range(?), type t: range(?)) {
   operator :(x: range(?), type t: string) {
     var ret: string;
 
-    if x.hasLowBound() then
+    if (x.boundedType == BoundedRangeType.bounded ||
+        x.boundedType == BoundedRangeType.boundedLow) then
       ret += x.lowBound:string;
     ret += "..";
-    if x.hasHighBound() {
+    if (x.boundedType == BoundedRangeType.bounded ||
+        x.boundedType == BoundedRangeType.boundedHigh) {
```

Comment:
Perhaps `x.hasDeclaredHighBound()` and ditto low bound?

Reply:
Nice idea.  Addressed this in 646ef84876a1096151f322932f29537bd9766ede.